src/overlays/actors/ovl_En_Mb/z_en_mb.c

```diff
@@ -134,170 +155,186 @@ static ColliderQuadInit sQuadInit = {
+    /* Unk arrow 3   */ DMG_ENTRY(2, ENMB_DMGEFF_DEFAULT),
+    /* Fire magic    */ DMG_ENTRY(0, ENMB_DMGEFF_FREEZE),
+    /* Ice magic     */ DMG_ENTRY(3, ENMB_DMGEFF_STUN_ICE),
+    /* Light magic   */ DMG_ENTRY(0, ENMB_DMGEFF_FREEZE),
+    /* Shield        */ DMG_ENTRY(0, ENMB_DMGEFF_IGNORE),
+    /* Mirror Ray    */ DMG_ENTRY(0, ENMB_DMGEFF_IGNORE),
+    /* Kokiri spin   */ DMG_ENTRY(1, ENMB_DMGEFF_DEFAULT),
+    /* Giant spin    */ DMG_ENTRY(4, ENMB_DMGEFF_DEFAULT),
+    /* Master spin   */ DMG_ENTRY(2, ENMB_DMGEFF_DEFAULT),
+    /* Kokiri jump   */ DMG_ENTRY(2, ENMB_DMGEFF_DEFAULT),
+    /* Giant jump    */ DMG_ENTRY(8, ENMB_DMGEFF_DEFAULT),
+    /* Master jump   */ DMG_ENTRY(4, ENMB_DMGEFF_DEFAULT),
+    /* Unknown 1     */ DMG_ENTRY(0, ENMB_DMGEFF_FREEZE),
+    /* Unblockable   */ DMG_ENTRY(0, ENMB_DMGEFF_IGNORE),
+    /* Hammer jump   */ DMG_ENTRY(4, ENMB_DMGEFF_DEFAULT),
+    /* Unknown 2     */ DMG_ENTRY(0, ENMB_DMGEFF_IGNORE),
 };
 
 static InitChainEntry sInitChain[] = {
     ICHAIN_S8(naviEnemyId, 74, ICHAIN_CONTINUE),
-    ICHAIN_F32_DIV1000(gravity, 64536, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -1000, ICHAIN_CONTINUE),
     ICHAIN_F32(targetArrowOffset, 5300, ICHAIN_STOP),
 };
 
+void EnMb_SetupAction(EnMb* this, EnMbActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
 void EnMb_Init(Actor* thisx, GlobalContext* globalCtx) {
     EnMb* this = THIS;
     s32 pad;
     Player* player = PLAYER;
-    s16 yawDiff;
+    s16 relYawFromPlayer;
 
     Actor_ProcessInitChain(&this->actor, sInitChain);
     ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 46.0f);
     this->actor.colChkInfo.mass = MASS_IMMOVABLE;
-    this->actor.colChkInfo.damageTable = &sDamageTable;
-    Collider_InitCylinder(globalCtx, &this->collider1);
-    Collider_SetCylinder(globalCtx, &this->collider1, &this->actor, &sCylinderInit);
-    Collider_InitTris(globalCtx, &this->collider3);
-    Collider_SetTris(globalCtx, &this->collider3, &this->actor, &sTrisInit, this->collider3Items);
-    Collider_InitQuad(globalCtx, &this->collider2);
-    Collider_SetQuad(globalCtx, &this->collider2, &this->actor, &sQuadInit);
+    this->actor.colChkInfo.damageTable = &sSpearMoblinDamageTable;
+    Collider_InitCylinder(globalCtx, &this->hitbox);
+    Collider_SetCylinder(globalCtx, &this->hitbox, &this->actor, &sHitboxInit);
+    Collider_InitTris(globalCtx, &this->frontShielding);
+    Collider_SetTris(globalCtx, &this->frontShielding, &this->actor, &sFrontShieldingInit, this->frontShieldingTris);
+    Collider_InitQuad(globalCtx, &this->attackCollider);
+    Collider_SetQuad(globalCtx, &this->attackCollider, &this->actor, &sAttackColliderInit);
 
     switch (this->actor.params) {
-        case -1:
-            SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gEnMbSkel_008F38, &gEnMbStandStillAnim, this->jointTable,
-                               this->morphTable, 28);
+        case -1: /* Spear Guard */
```

Comment:
Make an enum for params, I think, then a few of these comments can go (keep the default).